src/Serialization/containers.jl

```diff
@@ -145,6 +188,52 @@ function load_object(s::DeserializerState, T::Type{<:Matrix{S}}, params::NCRing)
   end
 end
 
+function load_object(s::DeserializerState, T::Type{<:Matrix{S}}, key::U) where {S, U <: Union{Int, Symbol}}
+  return load_node(s, key) do _
+    load_object(s, T)
+  end
+end
+
+
+################################################################################
+# Multidimensional Arrays
+@register_serialization_type Array "MultiDimArray"
+
+function save_object(s::SerializerState, arr::Array{T, N}) where {T, N}
+  arr_size = size(arr)
+  k = arr_size[N]
+  save_data_array(s) do
+    for t in 1:k
+      save_object(s, arr[[1:arr_size[i] for i in 1:N - 1]..., t])
+    end
```

Comment:
this is incredibly inefficient, as you are copying parts of the matrix over and over again. It would be better to implement this in a loop instead of using recursion.

Another note: This differs from https://github.com/oscar-system/Oscar.jl/blob/87141eb743d51ebee0eb03cbc7839840b92bbcdb/src/Serialization/containers.jl#L113 in the order of iteration. I think it would be best to keep that the same (aka in the outermost iteration, iterate over the first index)

Reply:
so you mean to flatten the array when storing? I agree this is more efficient, but it comes with the cost of losing the structure. Losing the structure is fine if we want to reload into oscar however in terms of long term data storage I would argue preserving the structure is something to take into consideration and is argueably more important. That being said if something becomes unserializable then sure, but we would need to add things to the file to make sure the the structure can be reconstructed. Currently with only the dimension of the array you would need to know the order of how julia iterates over the multidimensional array and you would need to store that order in the file.